src/main/kotlin/uk/co/poolefoundries/baldinggate/core/BaldingGateGame.kt

```diff
@@ -335,56 +333,53 @@ class BaldingGateGame : Game() {
-            var pos = positionMapper.get(enemy)
-            val movePositions = Array<PositionComponent>()
-
-            for (action in 0 until stats.currentAP) {
-                playerPos = positionMapper.get(player)
-                playerStats = statsMapper.get(player).stats
-
-                val actionPlan = SkeletonAI.getPlan(pos, playerPos, stats, playerStats)
-                when (actionPlan.actions.first()) {
-                    is MoveTowardsPlayer -> {
-                        //TODO move
-                        var tempPos = pos
-                        movePositions.add(tempPos)
-                        for (step in 0 until minOf(speed, pos.gridWiseDistance(playerPos)-1)) {
-                            tempPos += tempPos.direction(playerPos)
-                            movePositions.add(tempPos)
-                        }
-                        pos = tempPos
-
-                    }
-                    is AttackPlayer -> {
-                        player.add(
-                            StatsComponent(
-                                statsMapper.get(player).stats.copy(
-                                    hitPoints = playerStats.hitPoints - stats.attack.roll()
-                                )
-                            )
-                        )
-                        println("Hitpoints: " + statsMapper.get(player).stats.hitPoints)
-                    }
-                    is Win -> {
-                        player.remove(VisualComponent::class.java)
-                        println("HAHAHAHAHA YOU DIED!")
-                    }
+        // TODO: Add an ID component so we can identify entities by a UUID
+        val playerIds = players.associateBy { UUID.randomUUID().toString() }
+        val enemyIds = enemies.associateBy { UUID.randomUUID().toString() }
+
+        val actions = SkeletonAI.getPlan(playerIds.toMobInfo(), enemyIds.toMobInfo())
+
+        actions.actions.forEach {
+            when(it) {
+                is MoveTowards -> {
+                    // TODO: Add animations
+                    val enemy = enemyIds.getValue(it.selfId).toMobInfo(it.selfId)
+                    val target = playerIds.getValue(it.targetId).toMobInfo(it.targetId)
+
+                    val newPos = it.getNewPos(enemy, target)
+
+                    enemyIds.getValue(it.selfId).add(newPos)
+
+                }
+                is Attack -> {
+                    val enemy = enemyIds.getValue(it.selfId).toMobInfo(it.selfId)
+                    val target = playerIds.getValue(it.targetId).toMobInfo(it.targetId)
+
+                    val damage = enemy.stats.attack.roll()
+                    val newStats = target.stats.applyDamage(damage)
+                    playerIds.getValue(it.targetId).add(StatsComponent(newStats))
+
+                    println("Big oooof you just took $damage damage, ${newStats.hitPoints} hp left")
+                }
+                is Win -> {
+                    println("You is dead!!!")
```

Comment:
this is not actually necessarily true because they might roll a low roll and the player might still be alive but heyho

Reply:
I think this is actually when the player is dead because that action will only be valid if the player already has 0HP right? or does all AP get spent and the worldstate fed in before applying the actions?